espnet2/tasks/enh_s2t.py

```diff
@@ -0,0 +1,465 @@
+enh_separator_choices.name = "enh_separator"
+enh_criterions_choices = copy.deepcopy(enh_criterion_choices_)
+enh_criterions_choices.name = "enh_criterions"
+
+# ASR (also SLU)
+asr_preencoder_choices = copy.deepcopy(asr_preencoder_choices_)
+asr_preencoder_choices.name = "asr_preencoder"
+asr_encoder_choices = copy.deepcopy(asr_encoder_choices_)
+asr_encoder_choices.name = "asr_encoder"
+asr_postencoder_choices = copy.deepcopy(asr_postencoder_choices_)
+asr_postencoder_choices.name = "asr_postencoder"
+asr_decoder_choices = copy.deepcopy(asr_decoder_choices_)
+asr_decoder_choices.name = "asr_decoder"
+
+# ST
+st_preencoder_choices = copy.deepcopy(st_preencoder_choices_)
+st_preencoder_choices.name = "st_preencoder"
+st_encoder_choices = copy.deepcopy(st_encoder_choices_)
+st_encoder_choices.name = "st_encoder"
+st_postencoder_choices = copy.deepcopy(st_postencoder_choices_)
+st_postencoder_choices.name = "st_postencoder"
+st_decoder_choices = copy.deepcopy(st_decoder_choices_)
+st_decoder_choices.name = "st_decoder"
+st_extra_asr_decoder_choices = copy.deepcopy(st_extra_asr_decoder_choices_)
+st_extra_asr_decoder_choices.name = "st_extra_asr_decoder"
+st_extra_mt_decoder_choices = copy.deepcopy(st_extra_mt_decoder_choices_)
+st_extra_mt_decoder_choices.name = "st_extra_mt_decoder"
+
+MAX_REFERENCE_NUM = 100
+
+name2task = dict(
+    enh=EnhancementTask,
+    asr=ASRTask,
+    st=STTask,
+)
+
+# More can be added to the following attributes
+enh_attributes = [
+    "encoder",
+    "encoder_conf",
+    "separator",
+    "separator_conf",
+    "decoder",
+    "decoder_conf",
+    "criterions",
+]
+
+asr_attributes = [
+    "token_list",
+    "input_size",
+    "frontend",
+    "frontend_conf",
+    "specaug",
+    "specaug_conf",
+    "normalize",
+    "normalize_conf",
+    "preencoder",
+    "preencoder_conf",
+    "encoder",
+    "encoder_conf",
+    "postencoder",
+    "postencoder_conf",
+    "decoder",
+    "decoder_conf",
+    "ctc_conf",
+]
+
+st_attributes = [
+    "token_list",
+    "src_token_list",
+    "input_size",
+    "frontend",
+    "frontend_conf",
+    "specaug",
+    "specaug_conf",
+    "normalize",
+    "normalize_conf",
+    "preencoder",
+    "preencoder_conf",
+    "encoder",
+    "encoder_conf",
+    "postencoder",
+    "postencoder_conf",
+    "decoder",
+    "decoder_conf",
+    "ctc_conf",
+    "extra_asr_decoder",
+    "extra_asr_decoder_conf",
+    "extra_mt_decoder",
+    "extra_mt_decoder_conf",
+]
+
+
+class EnhS2TTask(AbsTask):
+    # If you need more than one optimizers, change this value
+    num_optimizers: int = 1
+
+    # Add variable objects configurations
+    class_choices_list = [
+        # --enh_encoder and --enh_encoder_conf
+        enh_encoder_choices,
+        # --enh_separator and --enh_separator_conf
+        enh_separator_choices,
+        # --enh_decoder and --enh_decoder_conf
+        enh_decoder_choices,
+        # --enh_criterion and --enh_criterion_conf
+        enh_criterions_choices,
```

Comment:
I guess this will only allow a single loss function for the ENH task. It is better to remove this from `class_choices_list` and add an argument `--criterions` in `add_task_arguments` as in [espnet2/tasks/enh.py](https://github.com/espnet/espnet/blob/master/espnet2/tasks/enh.py).

Reply:
done.